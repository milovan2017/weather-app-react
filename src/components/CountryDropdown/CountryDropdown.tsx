import React from "react";
import { Form } from "react-bootstrap";

interface ICountryDropdownProps {
  shortcodes: string[];
  setCountryCode: (code: string) => void;
  countryCode: string;
}

export const CountryDropdown: React.FC<ICountryDropdownProps> = ({
  shortcodes,
  setCountryCode,
  countryCode,
}) => {
  return (
    <>
      <Form.Select
        className="text-muted"
        onChange={(e) => {
          setCountryCode(e.target.value);
        }}
        value={countryCode}
        name="countrycodes"
      >
        {shortcodes.map((code, index) => (
          <option key={index} value={code}>
            {code}
          </option>
        ))}
      </Form.Select>
    </>
  );
};
