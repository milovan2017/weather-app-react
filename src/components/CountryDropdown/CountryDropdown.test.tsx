import { CountryDropdown } from "./CountryDropdown";
import { render, cleanup, screen } from "@testing-library/react";

afterEach(cleanup);

const mockShortcodes = ["rs", "ru", "gr"];
const mockCountrycode = "rs";

describe("CountryDropdown.tsx", () => {
  it("renders DisplayWeather without crashing", () => {
    const { container } = render(
      <CountryDropdown
        shortcodes={mockShortcodes}
        countryCode={mockCountrycode}
        setCountryCode={() => {}}
      />
    );
    expect(container).toMatchSnapshot();
  });

  it("should display the correct number of options", () => {
    render(
      <CountryDropdown
        shortcodes={mockShortcodes}
        countryCode={mockCountrycode}
        setCountryCode={jest.fn()}
      />
    );
    const options = screen.getAllByRole("option");
    expect(options.length).toBe(mockShortcodes.length);
  });
});
