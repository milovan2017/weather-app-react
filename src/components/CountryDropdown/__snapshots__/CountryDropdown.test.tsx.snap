// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CountryDropdown.tsx renders DisplayWeather without crashing 1`] = `
<div>
  <select
    class="text-muted form-select"
    name="countrycodes"
  >
    <option
      value="rs"
    >
      rs
    </option>
    <option
      value="ru"
    >
      ru
    </option>
    <option
      value="gr"
    >
      gr
    </option>
  </select>
</div>
`;
